src/controllers/dynakube/dynatraceclient/factory.go

```diff
@@ -0,0 +1,108 @@
+package dynatraceclient
+
+import (
+	"context"
+	"time"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/token"
+	"github.com/Dynatrace/dynatrace-operator/src/dtclient"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects/address"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const apiTokenProbeDelay = 5 * time.Minute
+
+type BuildFunc func(properties Properties) (dtclient.Client, error)
+
+type Factory struct {
+	ctx                 context.Context
+	client              client.Client
+	dynatraceClientFunc BuildFunc
+}
+
+func NewFactory(client client.Client, dtClientFunc BuildFunc) *Factory {
+	return &Factory{
+		client:              client,
+		dynatraceClientFunc: dtClientFunc,
+	}
+}
+
+func (r *Factory) Create(ctx context.Context, dynakube *dynatracev1beta1.DynaKube) (dtclient.Client, error) {
+	r.ctx = ctx
+	tokenReader := token.NewReader(r.client, dynakube)
+	tokens, err := tokenReader.ReadTokens(ctx)
+
+	if err != nil {
+		return nil, err
+	}
+
+	err = tokens.VerifyValues()
+
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Nit: I think it's more readable without newlines between `err` value assignment and then its check.
```suggestion

	err = tokens.VerifyValues()
	if err != nil {
		return nil, err
	}

```

Reply:
maybe that is one of the things we could write down in the contribution guidelines. Looking at it, I agree, if it's a single line followed by error handling it looks better if it is cuddled. if there are multiple lines there should be a whitespace imo.

E.g.:
```
err = function1()
if err != nil {
  doStuff()
}
```
```
value1 := assignment1()
value2 := assignment2()
err := function1()

if err != nil {
  doStuff()
}
```
Thoughts?

Reply:
I like it!